### PR TITLE
fix: Properly clean up when a flux immediately errors or completes

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

File: fusion-endpoint/src/test/java/dev/hilla/push/PushMessageHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+        return msg -> {
+        };
+    }
+
+    private Consumer<AbstractClientMessage> ignoreUpdateMessages() {
+        return ignore(ClientMessageUpdate.class);
+    }
+
+    private Consumer<AbstractClientMessage> ignore(Class<?>... toIgnore) {
+        List<AbstractClientMessage> unexpectedMessages = new ArrayList<>();
```
Comment (cromoteca) [inline fusion-endpoint/src/test/java/dev/hilla/push/PushMessageHandlerTest.java:297]: This list seems not to be accessed anywhere. There is another list with the same name in the class.

Comment (Artur-) [reply fusion-endpoint/src/test/java/dev/hilla/push/PushMessageHandlerTest.java]: This was fixed in the previous PR

**Comment by Artur-** [comment]
Rebased on top of master

File: fusion-endpoint/src/main/java/dev/hilla/push/PushMessageHandler.java [hunk 2]
```diff
@@ -123,31 +138,46 @@ private void handleSubscribe(SubscribeMessage message,
...
+                    getLogger().error("Exception in Flux", error);
+                });
             }, () -> {
                 // Flux completed
-                closeHandlers.remove(message.getId());
-                send(sender, new ClientMessageComplete(message.getId()));
+
+                // Ensure that the subscription data has been stored before it
+                // is used
+                waitForSubscriptionData.whenComplete((a, b) -> {
```
[the remark this answers is not in the document]
Comment (Artur-) [reply fusion-endpoint/src/main/java/dev/hilla/push/PushMessageHandler.java:153]: The Lift guys commented on this problem/suggestion:

"Indeed this is a false positive https://github.com/google/error-prone/issues/988."

**Review by platosha — APPROVED** [review]

**Pull request merged**